### PR TITLE
fix: fix the way of getBlockLimit and mofify function name

### PR DESCRIPTION
**Pull request by sulenn**

### CODE REVIEW

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ conn/channel.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6078806&bid=19900763)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ conn/channel.go  -2
         

Clones removed
==============
+ conn/channel.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6078806&bid=19900763)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ conn/channel.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6078806&bid=19900763)

File: client/api_handler.go [hunk 1]
```diff
@@ -297,13 +297,21 @@ func (api *APIHandler) GetBlockNumber(ctx context.Context, groupID int) ([]byte,
 
 // GetBlockLimit returns the blocklimit for current blocknumber
 func (api *APIHandler) GetBlockLimit(ctx context.Context, groupID int) (*big.Int, error) {
-	defaultNumber := big.NewInt(500)
+	var blockLimit *big.Int
+	if !api.IsHTTP() {
+		blockLimit = api.Connection.GetBlockLimit()
```
Comment (bxq2011hust) [inline client/api_handler.go:302]:
```suggestion
		blockLimit = api.Connection.GetBlockNumber()
```

File: conn/connection.go [hunk 2]
```diff
@@ -328,44 +332,44 @@ func (c *Connection) SubscribeTopic(topic string, handler func([]byte)) error {
...
 
-func (c *Connection) PushTopicDataToALL(topic string, data []byte) error {
+func (c *Connection) BroadcastAmopMsg(topic string, data []byte) error {
 	hc := c.writeConn.(*channelSession)
-	return hc.pushTopicDataToALL(topic, data)
+	return hc.broadcastAmopMsg(topic, data)
 }
 
-func (c *Connection) PushAuthTopicDataRandom(topic string, data []byte) error {
+func (c *Connection) SendAmopPrivateMsg(topic string, data []byte) error {
```
Comment (bxq2011hust) [inline conn/connection.go:365]:
```suggestion
func (c *Connection) SendAMOPPrivateMsg(topic string, data []byte) error {
```

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ conn/channel.go  -2
         

Clones removed
==============
+ conn/channel.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6078806&bid=19900763)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ conn/channel.go  -2
         

Clones added
============
- examples/amop/multicast_pub/publisher.go  2
- examples/amop_auth/unicast_pub/publisher.go  4
- examples/amop_auth/multicast_pub/publisher.go  5
- examples/amop/unicast_pub/publisher.go  2
- examples/amop_auth/sub/subscriber.go  1
         

Clones removed
==============
+ conn/channel.go  -2
+ precompiled/config/system_config.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6078806&bid=19900763)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ conn/channel.go  -2
         

Clones added
============
- examples/amop/multicast_pub/publisher.go  2
- examples/amop_auth/sub/subscriber.go  1
- examples/amop_auth/unicast_pub/publisher.go  4
- examples/amop_auth/multicast_pub/publisher.go  5
- examples/amop/unicast_pub/publisher.go  2
         

Clones removed
==============
+ precompiled/config/system_config.go  -2
+ conn/channel.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6078806&bid=19900763)

File: conn/connection.go [hunk 3]
```diff
@@ -701,3 +701,9 @@ func (c *Connection) read(codec ServerCodec) {
 func (c *Connection) IsHTTP() bool {
 	return c.isHTTP
 }
+
+// GetBlockLimit returns BlockLimit
+func (c *Connection) GetBlockNumber() int64 {
+	hc := c.writeConn.(*channelSession)
```
Comment (bxq2011hust) [inline conn/connection.go:707]: if？

File: client/api_handler.go [hunk 4]
```diff
@@ -40,6 +40,9 @@ type APIHandler struct {
 
 const (
 	indent = "  "
+
+	// other
```
Comment (bxq2011hust) [inline client/api_handler.go:44]:
```suggestion

```

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ conn/channel.go  -2
         

Clones added
============
- examples/amop_auth/multicast_pub/publisher.go  5
- examples/amop/unicast_pub/publisher.go  2
- examples/amop_auth/unicast_pub/publisher.go  4
- examples/amop/multicast_pub/publisher.go  2
- examples/amop_auth/sub/subscriber.go  1
         

Clones removed
==============
+ conn/channel.go  -2
+ precompiled/config/system_config.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6078806&bid=19900763)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ conn/channel.go  -2
         

Clones added
============
- examples/amop/multicast_pub/publisher.go  2
- examples/amop/unicast_pub/publisher.go  2
- examples/amop_auth/multicast_pub/publisher.go  5
- examples/amop_auth/unicast_pub/publisher.go  4
- examples/amop_auth/sub/subscriber.go  1
         

Clones removed
==============
+ conn/channel.go  -2
+ precompiled/config/system_config.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6078806&bid=19900763)

**Review by bxq2011hust — DISMISSED** [review]

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ conn/channel.go  -2
         

Clones added
============
- examples/amop/unicast_pub/publisher.go  2
- examples/amop_auth/multicast_pub/publisher.go  5
- examples/amop_auth/unicast_pub/publisher.go  4
- examples/amop_auth/sub/subscriber.go  1
- examples/amop/multicast_pub/publisher.go  2
         

Clones removed
==============
+ precompiled/config/system_config.go  -2
+ conn/channel.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=6078806&bid=19900763)

**Pull request merged**